Sources/xcodeproj/Objects/Project/PBXProj.swift

```diff
@@ -70,10 +70,22 @@ public final class PBXProj: Decodable {
         classes = try container.decodeIfPresent([String: Any].self, forKey: .classes) ?? [:]
         let objectsDictionary: [String: Any] = try container.decodeIfPresent([String: Any].self, forKey: .objects) ?? [:]
         let objectsDictionaries: [String: [String: Any]] = (objectsDictionary as? [String: [String: Any]]) ?? [:]
-        try objectsDictionaries.forEach { reference, dictionary in
-            let object = try PBXObject.parse(reference: reference, dictionary: dictionary, userInfo: decoder.userInfo)
+        
+        let dictionary = NSDictionary(dictionary: objectsDictionaries)
+        dictionary.enumerateKeysAndObjects(options: .concurrent) { (key, obj, _) in
+            guard let reference = key as? String,
+                let dictionary = obj as? Dictionary<String, Any>
+                else { return }
+            let userInfo = decoder.userInfo
+            let parser = PBXObjectParser(
```

Comment:
I'd move this outside `dictionary.enumerateKeysAndObjects` and change its constructor to be:

```swift
let parser = PBXObjectParser(userInfo: userInfo)
```

Then you can call it for each object:

```swift
parser.parse(reference: reference, dictionary: dictionary)
```